pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -748,6 +874,20 @@ public String toString(Object value) {
     return getSingleValueType().toString(toObjectArray(value)[0]);
   }
 
+  public String toJson(Object value) {
+    String jsonString = getSingleValueType().toString(toObjectArray(value)[0]);
+
+    // Check if the string is valid JSON.
+    JsonNode jsonNode = null;
+    try {
+      jsonNode = JsonUtils.stringToJsonNode(jsonString);
+    } catch (IOException ioe) {
+      throw new UnsupportedOperationException("Cannot convert value from STRING to JSON");
+    }
+
+    return jsonNode.toString();
+  }
```

Comment:
For string value, we should try to parse it as json string; for other type value, we should try to parse it as general object:
```suggestion
  public String toJson(Object value) {
    if (value instanceof String) {
      try {
        return JsonUtils.stringToJsonNode((String) value).toString();
      } catch (Exception e) {
        throw new RuntimeException("Caught exception while parsing string '" + value + "' as json", e);
      }
    } else {
      try {
        return JsonUtils.objectToString(value).toString();
      } catch (Exception e) {
        throw new RuntimeException("Caught exception while serializing object '" + value + "' as json", e);
      }
    }
  }
```

Reply:
No need to override this method for different data types

Reply:
Fixed.